src/d3a/models/strategy/__init__.py

```diff
@@ -144,7 +144,7 @@ def can_offer_be_posted(
 
         if replace_existing:
             # Do not consider previous offers, since they would be replaced by the current one
-            posted_offer_energy = 0.0
+            posted_offer_energy = self.sold_offer_energy(market.id)
```

Comment:
I think it would be good to update the comment above to document this change. 
It could be also great to write in the PR description the reason why the previous version was wrong, and what this change is fixing. This might help in the future.

Reply:
@fievelk the `posted_offer_energy` consists of open and traded offers, In the case of replace existing  we want to exclude the open offers but the take the sold ones into consideration